plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -49,23 +49,66 @@ import {
 import { TeamMember as AdoTeamMember } from 'azure-devops-node-api/interfaces/common/VSSInterfaces';
 import { Logger } from 'winston';
 import { PolicyEvaluationRecord } from 'azure-devops-node-api/interfaces/PolicyInterfaces';
-import { WebApi } from 'azure-devops-node-api';
+import {
+  WebApi,
+  getBearerHandler,
+  getPersonalAccessTokenHandler,
+} from 'azure-devops-node-api';
 import {
   TeamProjectReference,
   WebApiTeam,
 } from 'azure-devops-node-api/interfaces/CoreInterfaces';
 import { UrlReader } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  DefaultAzureDevOpsCredentialsProvider,
+  ScmIntegrations,
+} from '@backstage/integration';
 
 /** @public */
 export class AzureDevOpsApi {
-  public constructor(
-    private readonly logger: Logger,
-    private readonly webApi: WebApi,
-    private readonly urlReader: UrlReader,
-  ) {}
+  private readonly logger: Logger;
+  private readonly urlReader: UrlReader;
+  private readonly config: Config;
+
+  private constructor(logger: Logger, urlReader: UrlReader, config: Config) {
+    this.logger = logger;
+    this.urlReader = urlReader;
+    this.config = config;
+  }
+
+  static fromConfig(
+    config: Config,
+    options: { logger: Logger; urlReader: UrlReader },
+  ) {
+    return new AzureDevOpsApi(options.logger, options.urlReader, config);
+  }
+
+  private async getWebApi(host?: string, org?: string): Promise<WebApi> {
+    const validHost = host ?? this.config.getString('azureDevOps.host');
+    const validOrg = org ?? this.config.getString('azureDevOps.organization');
+    const url = `https://${validHost}/${validOrg}`;
+
+    const scmIntegrations = ScmIntegrations.fromConfig(this.config);
+    const credentialsProvider =
+      DefaultAzureDevOpsCredentialsProvider.fromIntegrations(scmIntegrations);
+    const credentials = await credentialsProvider.getCredentials({
+      url,
+    });
+
+    const authHandler =
```

Comment:
I came accross [getHandlerFromToken](https://github.com/microsoft/azure-devops-node-api/blob/55908ae28d367e7b30c167f679c4f1087f647cb2/api/WebApi.ts#L67) that figures out whether its a bearer token or a PAT.

Reply:
I played around with this but found it made the code messy. Until we removed the `azureDevOps.token` config value I think it's best to leave it as is. I did add a TODO about using this when that happens 👍